daffodil-core/src/main/scala/org/apache/daffodil/lib/util/Misc.scala

```diff
@@ -746,17 +746,12 @@ object Misc {
     }
   }
 
-  // helper function for scala 2.13 and 3 compatibility
+  /**
+   * scala 3 changes the way lazy val are deserialized so it looks like fieldName$lzy1
+   * This is a helper function to keep from dirtying the code
+   */
   def lookupDeclaredField(clazz: Class[?], fieldName: String): Field = {
-    try {
-      // TODO scala 2.12 phase out, we can remove the check for the plain name, and always
-      // check the lazy name version instead
-      clazz.getDeclaredField(fieldName)
```

Comment:
Should we keep this try/catch, just with a tweak to the comment?

This function is about using reflection to find a specific field of a class, so ideally it would work regardless if the field is lazy. With this change it only works for lazy fields.

Currently we only use it for `parents`, which is lazy and so everything works, but if we ever change `parents` to be non-lazy or use this function for some other non-lazy field it won't work as expected.